Lib/csv.py

```diff
@@ -409,14 +409,10 @@ def has_header(self, sample):
                 continue # skip rows that have irregular number of columns
 
             for col in list(columnTypes.keys()):
-
-                for thisType in [int, float, complex]:
-                    try:
-                        thisType(row[col])
-                        break
-                    except (ValueError, OverflowError):
-                        pass
-                else:
+                thisType = complex
+                try:
+                    thisType(row[col])
+                except:
```

Comment:
I guess a bare except is okay here since a __complex__ method could raise any exception. Since it's typically not considered good form, perhaps a comment would be useful? Just a thought though.

Reply:
I thought about it a bit more, my original idea was that any error here should fall back to str type, but now I think it could happen that in the future logic changes and makes it not ok to catch IndexError here (right now it wont' matter either way).

I've also looked at numeric types docs and they seem to say only ValueError or OverflowError are possible here, so that's good to know :)